scene/gui/control.cpp

```diff
@@ -449,6 +449,12 @@ void Control::_update_canvas_item_transform() {
 	Transform2D xform = _get_internal_transform();
 	xform[2] += get_position();
 
+	// We use a little workaround to avoid flickering when moving the pivot with _edit_set_pivot()
+	if (is_inside_tree() && Math::abs(Math::sin(data.rotation * 4.0f)) < 0.00001f && get_viewport()->is_snap_controls_to_pixels_enabled()) {
+		xform[2] = xform[2].round();
+		xform.set_rotation_and_scale(xform.get_rotation(), xform.get_scale().round());
```

Comment:
Please don't do that, the scale should not be rounded. 
Keep in mind the snap to pixel makes only sense with non-rotated and non-scaled nodes, so in other situations it should not alter the display.

Reply:
Yeah knew that, just wasn't paying enough attention, maybe letting r_margins[2] and r_margins[3] remain floored?